apps/intellij/src/main/kotlin/dev/nx/console/angular/NxAngularConfigService.kt

```diff
@@ -0,0 +1,109 @@
+
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.application.EDT
+import com.intellij.openapi.components.*
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.project.RootsChangeRescanningInfo
+import com.intellij.openapi.roots.ex.ProjectRootManagerEx
+import com.intellij.openapi.util.EmptyRunnable
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.openapi.vfs.VirtualFileManager
+import dev.nx.console.nxls.NxWorkspaceRefreshListener
+import dev.nx.console.nxls.NxlsService
+import dev.nx.console.nxls.NxlsService.Companion.NX_WORKSPACE_REFRESH_TOPIC
+import dev.nx.console.utils.writeAction
+import java.nio.file.Path
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import org.angular2.cli.config.AngularConfig
+import org.angular2.cli.config.AngularConfigProvider
+
+@Service(Service.Level.PROJECT)
+@State(name = "NxAngularConfigService", storages = [Storage("nx-angular-config.xml")])
+class NxAngularConfigService(private val project: Project, private val cs: CoroutineScope) :
+    PersistentStateComponent<NxAngularConfigService.AngularConfigState> {
+
+    init {
+        project.messageBus
+            .connect(cs)
+            .subscribe(NX_WORKSPACE_REFRESH_TOPIC, NxWorkspaceRefreshListener { refresh() })
+    }
+
+    var config: NxAngularConfig? = null
+
+    override fun getState(): AngularConfigState? =
+        config?.let { config ->
+            AngularConfigState(config.file.url, config.projectFiles.mapValues { it.value.url })
+        }
+
+    override fun loadState(state: AngularConfigState) {
+        val vfManager = VirtualFileManager.getInstance()
+        val workspaceFile = vfManager.findFileByUrl(state.workspaceLocation)
+        val projectFiles =
+            state.projects
+                .mapNotNull { (name, url) -> vfManager.findFileByUrl(url)?.let { Pair(name, it) } }
+                .toMap()
+        config = if (workspaceFile != null) NxAngularConfig(workspaceFile, projectFiles) else null
+        refresh()
+    }
+
+    private fun refresh() {
+        cs.launch { refreshState() }
+    }
+
+    private suspend fun refreshState() {
+        val workspace = NxlsService.getInstance(project).workspace()
+        if (workspace == null) {
+            updateConfig(null)
+            return
+        }
+        val manager = VirtualFileManager.getInstance()
+        val workspaceRoot = manager.findFileByNioPath(Path.of(workspace.workspacePath))
+        val workspaceConfig = workspaceRoot?.findFileByRelativePath("nx.json")
+        if (workspaceRoot == null || workspaceConfig == null) {
+            updateConfig(null)
+            return
+        }
+        val projectFiles =
+            workspace.workspace.projects.values
+                .asSequence()
+                // TODO use dependency graph here, or framework property
+                .filter { project ->
```

Comment:
@MaxKless So, this is the place where it's up to you now to improve the detection mechanism. Dependency graph is nice, but maybe you could add a property `framework` to the `project.json` spec, so that users can explicitly specify which framework they intend to use and get support for. These properties could be used both by WebStorm and VSCode to determine things like Vue or Angular Language Service startup.

If you add that property I will further improve the code, so that React and Vue frameworks are properly detected within the project.

Reply:
Okay, I will build something using the dependency graph here. I don't think a 'framework' property is an option here since it would be such a cross-cutting change. 

Appreciate the work you put in, especially making it async!!

Reply:
> I don't think a 'framework' property is an option here since it would be such a cross-cutting change

I would think of this property as an optional thing to specify, if auto-detection doesn't work. But I don't insist, WebStorm will have it's own solution here as well to facilitate situations when auto-detection fails.